src/components/NavigationSection.svelte

```diff
@@ -69,6 +76,22 @@
     }
   }
 
+  function getCategories (items) {
```

Comment:
I suggest to use `reduce` instead of `forEach`:
```js
function getCategorizedItems (items) {
  const categorizedItemsArray = items.reduce((accumulator, item) => {
    accumulator[item.category] = accumulator[item.category] || []      
    accumulator[item.category].push(item)
    return accumulator
   }, {})

  return Object.keys(categorizedItemsArray).map(key => {
    return {key: key, items: categorizedItemsArray[key]}
  })
}
```